### DISCUSSION TITLE
JSON schema-based documentation tooltip

### DISCUSSION THREAD

**Discussion by nikolaqm** [josdejong/svelte-jsoneditor#246] [Ideas]
It seems that currently there is no JSON schema-based tooltip showing description/title.
That was supported with [the old jsoneditor](https://github.com/josdejong/jsoneditor/pull/650).

Is that the case? Is there something that can be used for this purpose? Would it be hard to add?

**Comment by josdejong** [comment]
That's a good idea. In the new `svelte-jsoneditor`, you can implement custom value render components. We could implement one that reads the description from the JSON Schema, and if any, show an information icon with the description as tooltip. 

The [src/lib/plugins/value/renderJSONSchemaEnum.ts](https://github.com/josdejong/svelte-jsoneditor/blob/main/src/lib/plugins/value/renderJSONSchemaEnum.ts) plugin can serve as an example. 

Anyone interested in implementing a `renderJSONSchemaDescription` plugin?

**Comment by mgithubmessier** [comment]
I would love this feature -- this library has been fantastic and this is the only feature I could think of to improve it as well. It'd be awesome if it was available in the `vanilla-jsoneditor` npm package as well.
